doc/controller_configuration/controller_configuration_tutorial.rst

```diff
@@ -1,175 +1,498 @@
-         - joint_1
-         - joint_2
-         - joint_3
-         - joint_4
-         - joint_5
-         - joint_6
-     - name: fake_gripper_controller
-       joints:
-         []
+The ``type`` setting specifies the *fake controller type*:
 
-In order to load an initial pose, one can have a list of (group, pose) pairs as follows:
+* ``interpolate``: performs smooth interpolation between trajectory waypoints - the default for visualization.
+* ``via points``: jumps to the position specified by each trajectory waypoint without interpolation in between - useful for visual debugging.
+* ``last point``: warps directly to the last trajectory waypoint - the fastest method for off-line benchmarking.
 
-.. code:: yaml
+Multi Controller Manager
+^^^^^^^^^^^^^^^^^^^^^^^^
 
-   initial:
-     - group: arm
-       pose:  ready
+The *MoveIt Multi Controller Manager* can be used when more than one ``ros_control`` node is employed. It works by creating multiple *MoveIt ROS Control Controller Managers*, one for each node. It instantiates them with their respective namespace and takes care of proper delegation. This type of manager can be configured by setting ``moveit_controller_manager`` to ``moveit_ros_control_interface::MoveItMultiControllerManager``:
 
-Controller Switching and Namespaces
------------------------------------
+.. code-block:: XML
 
-All controller names get prefixed by the namespace of their ros_control node. For this reason, controller names should not contain slashes, and can't be named ``/``. For a particular node, MoveIt can decide which controllers to have started or stopped. Since only controller names with registered allocator plugins are handled over MoveIt, MoveIt takes care of stopping controllers based on their claimed resources if a to-be-started controller needs any of those resources.
+    <param name="moveit_controller_manager" value="moveit_ros_control_interface::MoveItMultiControllerManager" />
+
+Controller Switching and Namespaces
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Controllers for Multiple Nodes
-------------------------------
+All controller names get prefixed by the namespace of their ``ros_control`` node. For this reason, controller names should not contain slashes, and can’t be named ``/``.
 
-MoveItMultiControllerManager can be used for more than one ros_control nodes. It works by creating several MoveItControllerManagers, one for each node. It instantiates them with their respective namespace and takes care of proper delegation. To use it must be added to the launch file. ::
+Joints are claimed by each controller during initialization or startup. ROS controllers receive a hardware interface of the type they support (position, velocity, or effort), which they can use to request a *joint handle* for each joint they are configured to control. The act of requesting a joint handle will cause the *ROS Controller Manager* to record that a particular joint was *claimed* as a *resource* by a particular controller.
```

Comment:
This paragraph explains implementation details of the ros_control controller manager. This should not be part of this tutorial.

Reply:
I was on the fence about this too. It's one of the main reasons why people would come to this page (receiving an error about joints not being controlled by their controller). If you explain it, you're going too far and including extra information about other libraries. If you don't explain it, then you can't give someone an accurate picture of how controllers work with MoveIt. Just a weird gray area.

Is there a ROS Control 1 topic I can link to that states this precise thing? That would be splitting the difference. We make sure this is covered, but by linking to an external resource so the explanation stays with the product it belongs to.

Reply:
I don't know whether the ros_control tutorials explain that idea. Probably they do.

Reply:
I looked at ros_control tutorials and they don't explain this idea. That doesn't necessarily mean it needs to be included in MoveIt documentation (I could just explain this in a blog article instead, since it's "cross-cutting") but I recommend leaving it in.